### PR TITLE
fix: survey baseurl fix and addition of SHORT_SURVEY_BASE_URL in .env.example file

### PR DESCRIPTION
**Pull request by rohan9896**
## What does this PR do?

Fixes the SURVEY_BASE_URL and adds SHORT_SURVEY_BASE_URL  in .env.example file.

Fixes # (issue)
#1039

https://www.loom.com/share/9553e8f2ca424d71b3434e9d233a9246?sid=717857fa-1aaf-44b4-824a-83e62f3610b7

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by rohan9896** [comment]
> Hey @rohan9896, thanks a lot for the PR. Please try the following cases and let me know once all the below are handled:
> 
> * Put the SURVEY_BASE_URL to something like "http://localhost:3000/testme/"
> * Now start the app and see if you get this when you share a survey and if you can fill the survey and see the response on your dashboard.
> 
> If you get stuck at any step, then that means it a bug so we **should fix it in this PR** itself.
> 
> Please also add a check to throw an error if the Survey Base URL and the short survey URL are the same.

Sure, will be and update.

**Comment by rohan9896** [comment]
Hi @ShubhamPalriwala, I have added the check to throw an error if the Survey Base URL and the short survey URL are the same. here is the loom recording for that - https://www.loom.com/share/64398773d41041d4b227b6081eb0df2c?sid=f31fd685-41d6-4291-b6cb-6140b5713ccf

And, About this point `Put the SURVEY_BASE_URL to something like "http://localhost:3000/testme/"` - I don't think it will work as according to the directory structure of Next app, the survey will be opened inside the /s/[surveyId], so it must use /s/ only for survey links.
Screenshot 2023-10-10 at 3 22 24 PM

File: .env.example [hunk 1]
```diff
@@ -10,10 +10,10 @@
 
 WEBAPP_URL=http://localhost:3000
 
-SURVEY_BASE_URL=http://localhost:3000/i
+SURVEY_BASE_URL=http://localhost:3000/s
 
 # Set this if you want to have a shorter link for surveys
-SHORT_SURVEY_BASE_URL=
+SHORT_SURVEY_BASE_URL=http://localhost:3000/i
```
Comment (ShubhamPalriwala) [inline .env.example:16]: Please keep this empty be default as if it's empty, it wont be considered and the SURVEY_BASE_URL will work

File: apps/web/env.mjs [hunk 2]
```diff
@@ -47,8 +47,10 @@ export const env = createEnv({
     INVITE_DISABLED: z.enum(["1", "0"]).optional(),
     IS_FORMBRICKS_CLOUD: z.enum(["1", "0"]).optional(),
     VERCEL_URL: z.string().optional(),
-    SURVEY_BASE_URL: z.string().url().optional(),
-    SHORT_SURVEY_BASE_URL: z.string().url().optional().or(z.string().length(0)),
+    SURVEY_URL: z.object({
+      SURVEY_BASE_URL: z.string().url().optional(),
+      SHORT_SURVEY_BASE_URL: z.string().url().optional().or(z.string().length(0)),
+    }).refine((obj) => obj.SHORT_SURVEY_BASE_URL !== obj.SURVEY_BASE_URL),
```
Comment (ShubhamPalriwala) [inline apps/web/env.mjs:53]: Please remove the SURVEY_URL env as it creates confusion and we'd currently like to go with an easier approach

File: apps/web/env.mjs [hunk 3]
```diff
@@ -124,8 +126,10 @@ export const env = createEnv({
     NEXT_PUBLIC_POSTHOG_API_HOST: process.env.NEXT_PUBLIC_POSTHOG_API_HOST,
     FORMBRICKS_ENCRYPTION_KEY: process.env.FORMBRICKS_ENCRYPTION_KEY,
     VERCEL_URL: process.env.VERCEL_URL,
-    SURVEY_BASE_URL: process.env.SURVEY_BASE_URL,
-    SHORT_SURVEY_BASE_URL: process.env.SHORT_SURVEY_BASE_URL,
+    SURVEY_URL: {
+      SURVEY_BASE_URL: process.env.SURVEY_BASE_URL,
+      SHORT_SURVEY_BASE_URL: process.env.SHORT_SURVEY_BASE_URL,
+    },
```
Comment (ShubhamPalriwala) [inline apps/web/env.mjs:132]: same as above

File: packages/lib/constants.ts [hunk 4]
```diff
@@ -12,10 +12,12 @@ export const MAU_LIMIT = IS_FORMBRICKS_CLOUD ? 5000 : 1000000;
...
 
-export const SURVEY_BASE_URL = env.SURVEY_BASE_URL ? env.SURVEY_BASE_URL + "/" : `${WEBAPP_URL}/s/`;
+export const SURVEY_BASE_URL = env.SURVEY_URL.SURVEY_BASE_URL
+  ? env.SURVEY_URL.SURVEY_BASE_URL + "/"
+  : `${WEBAPP_URL}/s/`;
 
-export const SHORT_SURVEY_BASE_URL = env.SHORT_SURVEY_BASE_URL
-  ? env.SHORT_SURVEY_BASE_URL + "/"
+export const SHORT_SURVEY_BASE_URL = env.SURVEY_URL.SHORT_SURVEY_BASE_URL
+  ? env.SURVEY_URL.SHORT_SURVEY_BASE_URL + "/"
```
Comment (ShubhamPalriwala) [inline packages/lib/constants.ts:20]: same for these as well, please keep it as it is

**Review by ShubhamPalriwala — COMMENTED** [review]
Hey, so I went through your screen recording and comments, the best way forward w minimal changes is, for now, to just change the value of SURVEY_BASE_URL in the .env.example and get rid of anything that adds complexity as of now.

So please do that, test it thoroughly and ping me once its ready for review, we should then mostly be good to go

**Comment by rohan9896** [comment]
Hi @ShubhamPalriwala, Made all the necessary changes, please check.

**Review by ShubhamPalriwala — APPROVED** [review]

**Pull request merged**